src/utils/stream-response.js

```diff
@@ -14,21 +14,30 @@
  * limitations under the License.
  */
 
+import {Deferred} from './promise';
+
 /**
  * Decodes readable stream from response and writes to given writeable stream.
  * This function should be replaced with transform stream when well supported.
  * @param {!Window} win
  * @param {!Response} response
  * @param {!./detached-dom-stream.DetachedDomStream} writer
- * @return {!Promise}
+ * @return {!Promise<boolean>} Indicates if the response is empty.
```

Comment:
Can you clarify the doc here? Does this yield `true` when stream is empty or not empty? TBH, neither sounds very good from the API standpoint. But let's see if we can at least clarify it in docs.

Reply:
It returns `false` if empty. We need some signal if the response has content or not, but if you have concerns over the API I should probably change it?

Reply:
Bugged Dima offline :) we will leave for now, and can change if we think of a better API in future. Edited JSDoc to try to help clarify the behavior.